apps/ledger-live-desktop/src/renderer/IPCTransport.js

```diff
@@ -36,10 +38,29 @@ export class IPCTransport extends Transport {
   static isSupported = (): Promise<boolean> => Promise.resolve(typeof ipcRenderer === "function");
   // this transport is not discoverable
   static list = (): any => Promise.resolve([]);
-  static listen = () => {
+  static listen = observer => {
+    const requestId = uuidv4();
+    const replyChannel = `${transportListenChannel}_RESPONSE_${requestId}`;
+    const handler = (event, message) => {
+      if (message.error) {
+        observer.error(deserializeError(message.error));
+      } else {
+        const { data } = message;
+        if (data) {
+          observer.next(data);
+        } else {
+          observer.complete();
+        }
+      }
+    };
+
+    ipcRenderer.on(replyChannel, handler);
+    ipcRenderer.send(transportListenChannel, { requestId });
+
     return {
       unsubscribe: () => {
-        // empty fn
+        ipcRenderer.removeListener(replyChannel, handler);
+        ipcRenderer.send(transportListenUnsubscribeChannel, { requestId });
```

Comment:
Should we call the `unsubscribe` if the `observer` complete from the parent ?

Reply:
I was probably just thinking about Subjects and not Observable my bad, doesn't make sense if `observer` is a `Subscriber`